Build failure: `fahcontrol`

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `fahcontrol` on latest `nixos-unstable` (e4ad989506ec7d71f7302cc3067abd82730a4beb)

### Build log

```
error: builder for '/nix/store/8pv2wlc1iwrk87z0khg3ii4pdq44s9ir-python2.7-pycairo-1.18.2.drv' failed with exit code 1;
       last 10 log lines:
       >         with cairo.SVGSurface(None, 10, 10) as surface:
       > >           assert surface.get_document_unit() == cairo.SVGUnit.PT
       > E           AssertionError: assert cairo.SVGUnit.USER == cairo.SVGUnit.PT
       > E            +  where cairo.SVGUnit.USER = <built-in method get_document_unit of cairo.SVGSurface object at 0x7ffff7154c90>()
       > E            +    where <built-in method get_document_unit of cairo.SVGSurface object at 0x7ffff7154c90> = <cairo.SVGSurface object at 0x7ffff7154c90>.get_document_unit
       > E            +  and   cairo.SVGUnit.PT = <type 'cairo.SVGUnit'>.PT
       > E            +    where <type 'cairo.SVGUnit'> = cairo.SVGUnit
       >
       > tests/test_surface.py:262: AssertionError
       > =============== 1 failed, 258 passed, 4 skipped in 1.35 seconds ================
       For full logs, run 'nix log /nix/store/8pv2wlc1iwrk87z0khg3ii4pdq44s9ir-python2.7-pycairo-1.18.2.drv'.
error: 1 dependencies of derivation '/nix/store/yyh1nk3jggq6spsi96ml5ch1cqzb5qkj-python-2.7.18.7-env.drv' failed to build
error: 1 dependencies of derivation '/nix/store/966ck799lgf3n56dp05d5j23cr1drhf8-fahcontrol-7.6.21.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1rwrhg6cw7n25yj53wazgp96cy849y7g-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rxddci5p3r81b732sr2pmjkjamm3fj9n-nixos-system-leungbk-nixos-23.11.20231119.e4ad989.drv' failed to build
```

### Notify maintainers

@zimbatm

--- a/pkgs/development/python2-modules/pycairo/default.nix
+++ b/pkgs/development/python2-modules/pycairo/default.nix
@@ -1,5 +1,6 @@
 { lib
 , fetchFromGitHub
+, fetchpatch
 , meson
 , ninja
 , buildPythonPackage
@@ -21,6 +22,13 @@ buildPythonPackage rec {
     rev = "v${version}";
     sha256 = "142145a2whvlk92jijrbf3i2bqrzmspwpysj0bfypw0krzi0aa6j";
   };
+
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/pygobject/pycairo/commit/678edd94d8a6dfb5d51f9c3549e6ee8c90a73744.patch";
+      sha256 = "sha256-HmP69tUGYxZvJ/M9FJHwHTCjb9Kf4aWRyMT4wSymrT0=";
+    })
+  ];
 
   nativeBuildInputs = [
     meson

